src/admin/mod.rs

```diff
@@ -99,11 +101,11 @@ impl Server {
         self.addr
     }
 
-    pub fn spawn(self, shutdown: &signal::Shutdown, drain_rx: Watch) {
+    pub async fn spawn(self, drain_rx: Watch) {
         let _dx = drain_rx.clone();
         let ready = self.ready.clone();
         let workload_info = self.workload_info.clone();
-        let shutdown_trigger = shutdown.trigger();
+        let shutdown_trigger = self.shutdown_trigger.clone();
```

Comment:
In fact, there would be a new shutdown_trigger created when calling `shutdown.trigger()`, so I think it's unnecessary to call the `clone()` for it.

Reply:
We cannot if you look at L155

Reply:
yeah, how about to use another variable? I remember that the `trigger()` will clone a trigger.

Reply:
I think it is no difference.